src/Maestro/DependencyUpdater/DependencyUpdater.cs

```diff
@@ -79,27 +79,61 @@ public async Task StartUpdateDependenciesAsync(int buildId, int channelId)
         }
 
         /// <summary>
-        ///     Run a single subscription
+        ///     Run a single subscription, only accept the build Id specified 
         /// </summary>
         /// <param name="subscriptionId">Subscription to run the update for.</param>
+        /// <param name="buildId">BAR build id to run the update for</param>
+        /// 
+        /// <returns></returns>
+        public Task StartSubscriptionUpdateAsync(Guid subscriptionId, int buildId)
+        {
+            // This function and the one below could probably be combined, however it makes the queries far more readable / debuggable.
+            var subscriptionToUpdate = (from sub in Context.Subscriptions
```

Comment:
Does a test call this? This query is very complex, and there is a high chance it won't work in EF.

Also, can we not indent it so much? Continuation lines should just be indented one, they shouldn't be aligned so far to the right.

Reply:
It's a faked out version, and the controller I'm modifying only sticks things into the queue for this.  I've tested it locally using my super-complex debugging setup and it's a simpler version of the thing it's copying, so I'd expect it to work though.  I don't quite know how to rig up a test that hits this, but I'm at least adding coverage where there was absolutely none before...

Reply:
If you got it to run at least once locally that should be ok. My concern would be that the "let" and the "SingleOrDefault" wouldn't translate into SQL.

Reply:
Thanks.  I'll do it again after I finish this to-do list of feedbacks too.

Reply:
SingleOrDefault is supported: https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/ef/language-reference/supported-and-unsupported-linq-methods-linq-to-entities#paging-methods

Reply:
Now the hard part is reading 100 pages of specification to learn what the query expression syntax actually turns into. (Also, I really dislike the query expression syntax for that very reason)

Reply:
Tested again with latest commit (also note the old query has this too), as well as fixed indents to look nicer.